lte/gateway/python/magma/subscriberdb/client.py

```diff
@@ -0,0 +1,167 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import asyncio
+import datetime
+import logging
+
+import grpc
+from lte.protos.s6a_service_pb2 import DeleteSubscriberRequest
+from lte.protos.s6a_service_pb2_grpc import S6aServiceStub
+from lte.protos.subscriberdb_pb2 import (ListSubscribersRequest,
+                                         LTESubscription, SubscriberData)
+from magma.common.grpc_client_manager import GRPCClientManager
+from magma.common.rpc_utils import grpc_async_wrapper
+from magma.common.sdwatchdog import SDWatchdogTask
+from magma.common.service_registry import ServiceRegistry
+from magma.subscriberdb.store.sqlite import SqliteStore
+from magma.subscriberdb.metrics import SUBSCRIBER_SYNC_LATENCY, \
+    SUBSCRIBER_SYNC_SUCCESS_TOTAL, SUBSCRIBER_SYNC_FAILURE_TOTAL
+
+
+class SubscriberDBCloudClient(SDWatchdogTask):
+    """
+    SubscriberDBCloudClient for requesting subscribers from Orchestrator.
+
+    SubscriberDBCloudClient calls the Orchestrator's SubscriberDBCloud service
+    to fetch pages of subscribers stored in Orchestrator and updates the local
+    subscriberdb store with the changes.
+    """
+    DEFAULT_SUBSCRIBERDB_SYNC_INTERVAL = 180
```

Comment:
Aside from not liking defaults: looks like we're defining these defaults twice -- once in main and once in client